src/libs/HttpUtils.js

```diff
@@ -109,7 +118,12 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
     });
 
     const url = ApiUtils.getCommandURL({shouldUseSecure, command});
-    return processHTTPRequest(url, type, formData, data.canCancel);
+    return processHTTPRequest(url, type, formData, data.canCancel, command);
+}
+
+function cancelPendingReconnectAppRequest() {
```

Comment:
Why does this need it's own cancel controller?

Reply:
In some [edge cases](https://github.com/Expensify/App/issues/15627#issuecomment-1466419685) where we cannot re-order the `ReconnectApp` requests we should cancel them and re-queue them later. Basically the idea is to make sure `ReconnectApp` to always fire last.

For example if the update action was made immediately after switching online (i.e. `ReconnectApp` is already fired but not completed yet).

Reply:
>the idea is to make sure ReconnectApp always fires last

Yeah, again, it feels like we are making a weird exceptional case for `ReconnectApp`. IMO, there should be nothing special about `ReconnectApp`. The perception that it needs to "happen last" is entirely related to the data that it's returning. This change should have been held on a wider conversation with the team. It's sort of clear to me that the network layer isn't perfect and can't handle every edge case. But feels like we are hacking around a deficiency by breaking some core assumptions about how things work.